at-dungnguyen/Week4/2018_07_09_at_dungnguyen_exercise_day1_1.rb

```diff
@@ -0,0 +1,179 @@
+class News
+  def self.update(obj, title, content)
+    obj.title = title
+    obj.content = content
+  end
+
+  def self.delete(obj)
+    @store.each do |news|
+      @store.delete(news) if news[:id] == obj.id
+    end
+  end
+
+  def self.list
+    @store
+  end
+end
+
+class BbcNews < News
+  attr_accessor :id, :title, :content, :category
+  def initialize(id, title, content, category)
+    @id = id
+    @title = title
+    @content = content
+    @category = category
+  end
+
+  def self.save(obj)
+    h = { id: obj.id, title: obj.title, content: obj.content, category: obj.category }
+    @store ||= []
+    @store.push(h)
+  end
+end
+
+class CnnNews < News
+  attr_accessor :id, :title, :content, :publication_date
+  def initialize(id, title, content, publication_date)
+    @id = id
+    @title = title
+    @content = content
+    @publication_date = publication_date
+  end
+
+  def self.save(obj)
+    h = { id: obj.id, title: obj.title, content: obj.content, publication_date: obj.publication_date }
+    @store ||= []
+    @store.push(h)
+  end
+end
+
+class User
+  attr_accessor :id, :name
+  attr_reader :friends, :message
+  def initialize(id, name)
+    @id = id
+    @name = name
+    @organisation = []
+    @friends = []
+    @message = []
+  end
+
+  def subscribe(organisation)
+    @organisation.push(organisation)
+  end
+
+  def unsubscribe(organisation)
+    @organisation.delete(organisation)
+  end
+
+  def subscription_list
+    p @organisation
+  end
+
+  def show_news(organisation)
+    p 'No result' unless @organisation.include?(organisation)
+    @organisation.each do |obj|
+      p obj.list if obj == organisation
+    end
+  end
+
+  def add_friend(user)
+    @friends.push(user)
+    user.friends.push(self)
+  end
+
+  def unfriend(user)
+    @friends.delete(user)
+    user.friends.delete(self)
+  end
+
+  def list_friends
+    count = Hash.new(0)
+    @friends.each do |fr|
+      count[fr.name] += 1
+      if count[fr.name] > 1
+        p fr.name += "(#{fr.id})"
+      else
+        p fr.name
+      end
+    end
+  end
+
+  def send_message(friend, message)
+    if @friends.include?(friend)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/9a5e9bd9b4f9995c0cfc6487a01e9352bd3dfcac/at-dungnguyen/Week4/2018_07_09_at_dungnguyen_exercise_day1_1.rb#L103"></span>Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals)</td>
    </tr>
  </tbody>
</table>